cli/engine/providers/azure/InfrastructureBuilder.py

```diff
@@ -24,6 +24,23 @@ def __init__(self, docs, manifest_docs=[]):
         self.docs = docs
         self.manifest_docs = manifest_docs
 
+        # Check if there is a hostname_domain_extension we want to apply.
+        self.hostname_domain_extension = None
+        if self.cluster_model.specification.cloud.hostname_domain_extension != '':
+            self.hostname_domain_extension = self.cluster_model.specification.cloud.hostname_domain_extension
```

Comment:
Seems to be not needed or just 

```
self.hostname_domain_extension = self.cluster_model.specification.cloud.hostname_domain_extension
```

Reply:
Agreed, we can just use '' for the second check needed inside the vm construction.